plugins/terminal.koplugin/main.lua

```diff
@@ -93,28 +93,36 @@ local Terminal = WidgetContainer:extend{
 }
 
 function Terminal:isExecutable(file)
-    if os.execute(string.format("test -x %s", file)) == 0 then -- full path
-        return true
-    elseif os.execute(string.format("which %s 2>/dev/null 1>/dev/null", file)) == 0 then
-        return true
-    end
+    -- check if file is an executable or a command in PATH
+    return os.execute(string.format("test -x %s || command -v %s", file, file)) == 0
 end
 
 -- Try SHELL environment variable and some standard shells
 function Terminal:getDefaultShellExecutable()
     if self.default_shell_executable then return self.default_shell_executable end
 
-    local shell = {"mksh", "ksh", "zsh", "ash", "dash", "sh", "bash"}
-    table.insert(shell, os.getenv("SHELL"))
+    local shell = {
+        "bash",
+        "sh",
+        "dash",
```

Comment:
Putting dash higher than zsh or something seems evil (though in principle that should come from `SHELL`).

Reply:
The thought crossed my mind, and then I thought, eeeh, tiny devices, zsh might not be a good fit.

And then I just remembered I *ship* zsh on those and actually use it ;D.

Reply:
Okay, kept it below ash because we suck at dealing with RPROMPTs (I suppose?), making the whole thing super unwieldy when using zsh ;).

Reply:
Alright. /shrug